File: 371SumOfTwoIntegers.py
class Solution(object):
    def getSum(self, a, b):
        """
        :type a: int
        :type b: int
        :rtype: int
        """
      	MAX_INT = 0x7FFFFFFFF
      	MASK = 0x100000000
      	while b != 0:
      		temp = a & b
      		a = (a ^ b) % MASK
      		b = (temp << 1)% MASK
      	return a if a<= MAX_INT else a | (~MASK + 1)

File: 371SumOfTwoIntegers.cpp
/*
371. Sum of Two Integers

Calculate the sum of two integers a and b, but you are not allowed to use the operator + and -.

Example:
Given a = 1 and b = 2, return 3.

*/

class Solution{
public:
	int getSum(int a, int b){
		if (b == 0)
			return a;
		while (a != 0){
			int temp = (a^b);
			b = (a & b)<<1;
			a = temp;
		}
		return b;
	}
}